src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs

```diff
@@ -483,6 +495,245 @@ public void CheckHistoryPrevious()
         }
 
         [Fact]
+        public void CheckHistoryPreviousNotCircular()
+        {
+            //submit, submit, up, up, up
+            const string inputString1 = "1 ";
+            const string inputString2 = "2 ";
+            Window.InsertCode(inputString1);
```

Comment:
@amcasey AddInput does a bunch of other processing like making the buffer as readonly. So if the command is not actually submitted then the next call to AddInput fails. if we actually submit the command then it gets added to history twice - once in AddInput and once as part of executing the command. AddInput is doing much more than it's name suggests - we should perhaps rename it or refactor it.